File: implementation/common/src/main/java/com/ase/common/attendance/AttendeeEventList.java
package com.ase.common.attendance;

import java.io.Serializable;
import java.util.List;

/**
 * The AttendeeEventList class represents a list of event IDs associated with an attendee.
 * It is used to store and retrieve the list of events attended by an attendee.
 */
public record AttendeeEventList(String userID, List<String> eventIDs) implements Serializable {

}

File: implementation/frontend/frontend/src/dtos/attendeeEventList.ts
/**
 * Class representing a list of event IDs that a particular user is attending.
 */
export class AttendeeEventList {
  userID: string;
  eventIDs: string[];

  constructor(userID: string, eventIDs: string[]) {
    this.userID = userID;
    this.eventIDs = eventIDs;
  }
}
